src/main/scala/org/lyranthe/fs2_mongodb/MongoCollectionEffect.scala

```diff
@@ -0,0 +1,313 @@
+package org.lyranthe.fs2_mongodb
+
+import cats.effect.Async
+import cats.syntax.functor.toFunctorOps
+import com.mongodb.MongoNamespace
+import com.mongodb.async.client.MongoCollection
+import com.mongodb.bulk.BulkWriteResult
+import com.mongodb.client.model._
+import com.mongodb.client.result._
+import org.bson.conversions.Bson
+
+import scala.collection.JavaConverters._
+
+private[fs2_mongodb] final class MongoCollectionEffect[F[_], A](
```

Comment:
I've been thinking about this value class, and I'm not completely sure if no instantiation occurs when using it. Because, since this is created in an extension method of another _(implicit)_  value class.

I'll try to make some research about that, cause if when can't avoid the instantiaton of this class, it would be better if we include the implicit `Async[F]` as a class constructor argument.

Reply:
@fiadliel I just found [this](https://developer.lightbend.com/blog/2018-11-01-the-scala-2.12-2.13-inliner-and-optimizer/index.html) and it seems that adding the `@inline` annotation doesn't have any effect.

> "The @inline annotation only has an effect if the inliner is enabled. It tells the inliner to always try to inline the annotated method or callsite".

Since, this a library I think is a bad idea to activate the inliner, and probably most users won't do it either. Thus, we should just leave the code as it is, and hope that the **JIT** will optimize the calls.

So, I think is better to remove all `@inline` annotations, they add _clutter_ to the code.
What do you think ?